crates/state/src/batch.rs

```diff
@@ -252,3 +261,54 @@ impl BootstrapState {
         }
     }
 }
+
+#[derive(Clone, Debug, PartialEq, Eq, BorshDeserialize, BorshSerialize)]
+pub struct CheckpointProofOutput {
+    pub batch_info: BatchInfo,
+    pub bootstrap_state: BootstrapState,
+}
+
+impl CheckpointProofOutput {
+    pub fn new(batch_info: BatchInfo, bootstrap_state: BootstrapState) -> CheckpointProofOutput {
+        Self {
+            batch_info,
+            bootstrap_state,
+        }
+    }
+}
+
+#[derive(
+    Clone, Debug, PartialEq, Eq, Arbitrary, BorshSerialize, BorshDeserialize, Serialize, Deserialize,
+)]
+pub struct CommitmentInfo {
```

Comment:
See above comment, should reason about what this is really intending to represent beyond just being a container for the data.